res/controllers/hid-controller-api.d.ts

```diff
@@ -31,18 +31,18 @@ declare namespace controller {
      *  @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use
      *  @return Returns report data with ReportID byte as prefix
      */
-    function getInputReport(reportID:number):Uint8Array;
+    function getInputReport(reportID: number): Uint8Array;
```

Comment:
Can you double-check whether `getInputReport` actually returns an `Uint8Array`, I don't want to repeat the debacle we had recently with the type-error. Same for `getFeatureReport`. Thanks.
https://github.com/mixxxdj/mixxx/pull/4759#discussion_r894022894

Reply:
I added some inspection code 
```
if (data.constructor === Uint8Array) {
            console.error("data is Uint8Array");
        } 
        if (data.constructor === Uint8ClampedArray) {
            console.error("data is Uint8ClampedArray");
        } 
        if (data.constructor === Int8Array) {
            console.error("data is Int8Array");
        } 
        if (ArrayBuffer.isView(data)) {
            console.error("data is ArrayBuffer view");
        } 
        if (data.constructor === ArrayBuffer) {
            console.error("data is ArrayBuffer");
        } 
        if (Array.isArray(data)) {
            console.error("data is Array");
        }
```
and both functions return an ArrayBuffer, which is also documented this way: https://doc.qt.io/qt-5/qtqml-cppintegration-data.html#qbytearray-to-javascript-arraybuffer

But the often called .incomingData callback uses QJSValueList which is detected as Uint8Array (and ArrayBuffer view):
https://github.com/mixxxdj/mixxx/blob/f36cd4f29086aa5370ece5a10d198f690a3619da/src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp#L183-L200

Reply:
I'm sorry, I don't think I follow. so `getInputReport` does actually return an ArrayBuffer? how is that related to `ControllerScriptEngineLegacy::handleIncomingData`?

Reply:
The parsePacket function of the common-hid-parser is called either with `args` from controllername`.incomingData` (executed from `ControllerScriptEngineLegacy::handleIncomingData`) or with the return value from `getInputReport`. I just checked the type of both.

Reply:
Oh, so it gets passed different types? Or are we forced to pass it a Uint8Array to not break existing code?

Reply:
Originally it was the same type, but was changed intentionally in #4521

Reply:
right... this looks correct to me. So in the PR it was changed from returning a plain array to returning a QByteArray (so an `ArrayBuffer`). So your docs `function getInputReport(reportID: number): Uint8Array;` are wrong, or am I missing something?

Reply:
Yes, I need to fix the return type for these two functions in the docu.

Reply:
Thank you, that's all I really wanted to get to the bottom off. I'm sorry for causing you more work than necessary.